components/Table/TableEditButton.tsx

```diff
@@ -34,8 +67,83 @@ export default function TableEditButton({ rowData, columns }) {
   const handleOpen = () => setOpen(true);
   const handleClose = () => setOpen(false);
 
-  const handleEdit = () => {
-    handleClose();
+  const handleEdit = async (e) => {
+    e.preventDefault();
+    const listOfBlanks = [];
+    if (editedData['Barcode ID'] === '') {
+      listOfBlanks.push('barcode');
+    }
+    if (editedData.Title === '') {
+      listOfBlanks.push('title');
+    }
+    if (editedData.Author === '') {
+      listOfBlanks.push('author');
+    }
+    if (listOfBlanks.length > 0) {
+      setErrorMessage(
+        `The following fields cannot be blank: ${listOfBlanks.join(', ')}`
+      );
+      return;
+    }
+
+    // if barcode exists in the data, AND the barcode does not equal the selected row's barcode.....
+    const duplicateBarcode = data.filter(
+      (item) => item['Barcode ID'] === editedData['Barcode ID']
+    );
+    if (
+      duplicateBarcode.length > 0 &&
+      editedData['Barcode ID'] !== rowData['Barcode ID']
+    ) {
+      setErrorMessage('This barcode already exists.');
+      return;
+    }
+    const filteredData = data.filter((item) => item.id !== rowData.id);
+
+    setButtonText('Saving...');
+    const patchData = {
+      barcodeId: editedData['Barcode ID'],
+      title: editedData.Title,
+      author: editedData.Author,
+      genres: selectedGenres.filter((item) => item !== ''),
+      regions: selectedRegions.filter((item) => item !== ''),
+      location: selectedLocation
+    };
+
+    const url =
+      process.env.NODE_ENV === 'production'
+        ? 'http://localhost:3000'
+        : 'http://localhost:3000';
```

Comment:
This could also be swapped for `getEnvUrl()`

Reply:
Swapped it out with getEnvUrl()